crates/matrix-sdk/src/event_cache/linked_chunk.rs

```diff
@@ -25,6 +25,66 @@ use std::{
     },
 };
 
+pub trait LinkedChunkListener<Item, Gap>: Send + Sync {
+    type Error: fmt::Debug;
+
+    fn new_chunk_items(
+        &self,
+        previous: Option<ChunkIdentifier>,
+        new: ChunkIdentifier,
+        next: Option<ChunkIdentifier>,
+    ) -> Result<(), Self::Error>;
+
+    fn new_chunk_gap(
+        &self,
+        previous: Option<ChunkIdentifier>,
+        new: ChunkIdentifier,
+        next: Option<ChunkIdentifier>,
+        gap: &Gap,
+    ) -> Result<(), Self::Error>;
+
+    fn remove_chunk(&self, chunk: ChunkIdentifier) -> Result<(), Self::Error>;
+
+    fn insert_items(&self, at: Position, items: &[Item]) -> Result<(), Self::Error>;
+
+    fn truncate_items(&self, chunk: ChunkIdentifier, length: usize) -> Result<(), Self::Error>;
+}
```

Comment:
Can you doc-comment all of this, please?

Also nit: I think it would read better if `new_chunk_items` and `new_chunk_gap` had the `chunk` at the end: `new_items_chunk` / `new_gap_chunk`. What do you think?

Reply:
The trait has been removed, as discussed.